crates/node/src/service.rs

```diff
@@ -416,16 +416,34 @@ pub fn new_full(
         let settlement_provider: Box<dyn SettlementProvider> = match layer_kind {
             SettlementLayer::Ethereum => {
                 let ethereum_conf = EthereumConfig::try_from(&config_path)?;
-                Box::new(EthereumClient::try_from(ethereum_conf).map_err(|e| ServiceError::Other(e.to_string()))?)
+                Box::new(
+                    LocalStarknetContractClient::try_from(ethereum_conf)
+                        .map_err(|e| ServiceError::Other(e.to_string()))?,
+                )
+            }
+            #[cfg(feature = "mockup-settlement-provider")]
+            SettlementLayer::Mockup => {
+                Box::new(mc_settlement::mockup::MockupSettlementProvider { file_path: config_path })
             }
-            SettlementLayer::Mockup => Box::new(MockupSettlementProvider { file_path: config_path }),
+            _ => unreachable!("Unexpected settlement layer: {:?}", layer_kind),
```

Comment:
This souldn't be required as you are doing  match? You get an ettor?

Reply:
Since mockup is not feature-enabled, I marked the enum as #[non_exhaustive] (afaik this is considered as good practice in such cases). Thus we need to handle _

Reply:
https://doc.rust-lang.org/reference/attributes/type_system.html

It seems that it is used to guarantee backward compatibility downstream. I don't think this is something we care about.